geode-web-api/src/main/java/org/apache/geode/rest/internal/web/controllers/support/CacheProviderImpl.java

```diff
@@ -26,4 +28,13 @@
   public InternalCache getInternalCache() {
     return GemFireCacheImpl.getExisting();
```

Comment:
so here, we can return the InternalCacheForClientAccess object. Looks like all the caller of this method needs it for client access in the controller except for getting the querystore. We can add a method in the InternalCacheForClientAccess object to get the queryStore since looks like we should allow that for client access.